cmd/cspc-operator/app/storagepool_create.go

```diff
@@ -395,8 +395,15 @@ func getResourceRequirementForCStorPool(cspi *apis.CStorPoolInstance) *corev1.Re
 	return resourceRequirements
 }
 
-func getAuxResourceRequirement(cspi *apis.CStorPoolInstance) corev1.ResourceRequirements {
-	return cspi.Spec.AuxResources
```

Comment:
are we trying NOT to use AuxResources of cspi.Spec? If so, better to remove from the struct itself

Reply:
`getAuxResourceRequirement` is used to populate the `resourceRequirements` for sidecar containers. The cspi is populated using either using the CSPC `PoolConfig` `AuxResources` or the `DefaultAuxResources`. If both of them are nil then an empty `resourceRequirements` needs to be set on the sidecar containers.